code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+			higherSwap()
+			return
+	if(mindTypeTwo == "Artificial" && mindTypeOne != "Artificial")
+		if(scanRatingTwo <7)
+			errorMessage = "Scanners insufficient"
+			swapProgress = 0
+			currentlySwapping = FALSE
+			connectOne.icon_state = "mind_pod_closed"
+			connectTwo.icon_state = "mind_pod_closed"
+			icon_state = "mind_hub"
+			return
+		if(mindTypeTwo == "Lower")
+			animorphsSwap()
+			return
+		if(mindTypeOne == "Higher" && theFly.len == 0)
+			higherSwap()
+			return
+	else
+		errorMessage = "Unknown interruption"
+		currentlySwapping = FALSE
+		swapProgress = 0
+		unlockPods()
+
+/obj/machinery/mind_machine/mind_machine_hub/emp_act()
+	if(currentlySwapping)
+		malfSwap = TRUE
+		return
+
+/obj/machinery/mind_machine/mind_machine_hub/emag_act()
+	if(currentlySwapping)
+		spark(src)
+		malfSwap = TRUE
+		return
+
+//Procs for the mind swap//////////
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/lowerSwap(var/mob/living/LO, var/mob/living/LT)
+	LO = occupantOne
+	LT = occupantTwo
+	if(isanimal(LO) && isanimal(LT))
+		var/mob/living/simple_animal/SO = LO
+		var/mob/living/simple_animal/ST = LT
+		var/occOneFaction = SO.faction
+		var/occTwoFaction = ST.faction
+		SO.faction = occTwoFaction //Make friendly spiders and angry corgis
+		ST.faction = occOneFaction
+		var/occOneSpeakChance = SO.speak_chance
+		var/occTwoSpeakChance = ST.speak_chance
+		SO.speak_chance = occTwoSpeakChance
+		ST.speak_chance = occOneSpeakChance
+		var/occOneName = SO.name
+		var/occTwoName = ST.name
+		SO.name = occTwoName
+		ST.name = occOneName
+		var/list/occOneSpeak = list()
+		var/list/occTwoSpeak = list()
+		for(var/V in SO.speak)
+			occOneSpeak += V
+			SO.speak -= V
+		for(var/V in ST.speak)
+			occTwoSpeak += V
+			ST.speak -= V
+		for(var/V in occTwoSpeak)
+			SO.speak += V
+		for(var/V in occOneSpeak)
+			ST.speak += V
+		var/occOnePet = ST.is_pet
+		var/occTwoPet = SO.is_pet
+		SO.is_pet = occTwoPet
+		ST.is_pet = occOnePet
+	else
+		errorMessage = "Sum intelligence too low" //There's just nothing to swap for monkeymen
+		bluespaceConduit += 1
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	swapProgress = 0
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/higherSwap(var/mob/HO, var/mob/HT)
+	HO = occupantOne
+	HT = occupantTwo
+	if(HO.mind.special_verbs.len)	//Swapping players, stolen from mind transfer
+		for(var/V in HO.mind.special_verbs)
+			HO.verbs -= V
+	if(HT.mind.special_verbs.len)
+		for(var/V in HT.mind.special_verbs)
+			HT.verbs -= V //Simple animals can't have special verbs so we just remove them without re-adding. Admins fear the machine.
+	var/list/HO_spells = HO.spell_list.Copy()
+	var/list/HT_spells = HT.spell_list.Copy()
+	for(var/spell/S in HO.spell_list)
+		HO.remove_spell(S)
+	for(var/spell/S in HT.spell_list)
+		HT.remove_spell(S)
+	var/mob/living/dummy = new(src.loc)
```

Comment:
```suggestion
	var/mob/living/dummy = new(src)
```
You can avoid having the dummy ever actually appear in any form in realspace and save a bit since you are deleting it by spawning it into the machine itself. I feel liek there may be a better way to save on the spell transfers but idk.